tests/system/examples/dask/test_dask.py

```diff
@@ -110,3 +113,58 @@ def dask_pipe(x=1, y=10):
 
         # remove compiled dask.yaml file
         os.remove("daskpipe.yaml")
+
+    def test_dask_close(self):
+        self._logger.info("Initializing dask cluster")
+        cluster_start_time = datetime.datetime.now()
+
+        # initialize the dask cluster and get its dashboard url
+        client = self.dask_function.client
+        time_took = (datetime.datetime.now() - cluster_start_time).seconds
+        self._logger.info(
+            "Dask cluster initialization completed", took_in_seconds=time_took
+        )
+
+        worker_start_time = datetime.datetime.now()
+        client.wait_for_workers(self.dask_function.spec.replicas)
+        time_took = (datetime.datetime.now() - worker_start_time).seconds
+        self._logger.info("Workers initialization completed", took_in_seconds=time_took)
+
+        self._logger.info("Shutting Down Cluster")
+        self.dask_function.close()
+
+        # wait for the dask cluster to completely shut down
+        mlrun.utils.retry_until_successful(
+            5,
+            40,
+            self._logger,
+            True,
+            wait_for_dask_cluster_to_shutdown,
+            "mydask",
+        )
+
+        # Client supposed to be closed
+        with pytest.raises(AttributeError):
+            client.list_datasets()
+
+        # Cluster supposed to be decommissioned
+        with pytest.raises(RuntimeError):
+            client.restart()
+
+
+def wait_for_dask_cluster_to_shutdown(dask_cluster_name):
+    resources = mlrun.get_run_db().list_runtime_resources()
+    pods = resources[0].resources.pod_resources
+    pod_counter = 0
+    for pod in pods:
+        if pod.name.startswith(f"mlrun-{dask_cluster_name}"):
```

Comment:
Instead of filtering the pod name (duplicating the "knowledge" of the relation between pod name and function name) Use the params of `list_runtime_resources`